src/DiscordMember.js

```diff
@@ -71,8 +71,30 @@ class DiscordMember {
    * @returns {string} The formatted nickname
    * @memberof DiscordMember
    */
-  getNickname (data) {
-    return Util.formatDataString(this.discordServer.getSetting('nicknameFormat'), data, this.member)
+  async getNickname (data) {
+    let nicknameData = {
+      robloxUsername: data.robloxUsername,
+      robloxId: data.robloxId,
+      groupRank: null,
+      discordId: data.discordId,
+      discordName: data.discordName
+    }
+
+    if (this.discordServer.getSetting('nicknameGroup')) {
+      let apiRank = await DiscordServer.getRobloxMemberGroups(nicknameData.robloxId)
+
+      for (let groups of apiRank) {
+        if (parseInt(groups.Id) === parseInt(this.discordServer.getSetting('nicknameGroup'))) {
+          nicknameData.groupRank = groups.Role
+          break
+        } else {
+          nicknameData.groupRank = 'Guest'
```

Comment:
This is being repeatedly set every iteration. Remove the else block here and put after the for loop:

```js
nicknameData.groupRank = nicknameData.groupRank || 'Guest'